pretty_midi/containers.py

```diff
@@ -179,23 +179,29 @@ def __str__(self):
 
 
 class Lyric(object):
-    """Timestamped lyric text.
+    """Timestamped lyric text with corresponding note.
 
     Attributes
     ----------
     text : str
         The text of the lyric.
     time : float
         The time in seconds of the lyric.
+    note: :class:`pretty_midi.Note` object
+        The note that corresponds to lyric, None for empty text.
     """
 
     def __init__(self, text, time):
         self.text = text
         self.time = time
+        self.note = None
+
+    def set_note(self, note):
+        self.note = note
 
     def __repr__(self):
-        return 'Lyric(text="{}", time={})'.format(
-            self.text.replace('"', r'\"'), self.time)
+        return 'Lyric(text="{}", time={}, note={})'.format(
+            self.text.replace('"', r'\"'), self.time, self.note)
```

Comment:
If you use self.note in this format string, it will use self.note.__str__() which will make the __repr__ of Lyric invalid. You should use self.note.__repr__() here instead.

Reply:
You are right, noted